src/libraries/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509Certificate2.cs

```diff
@@ -926,12 +972,39 @@ public static X509Certificate2 CreateFromEncryptedPem(ReadOnlySpan<char> certPem
                 {
                     Oids.Rsa => ExtractKeyFromEncryptedPem<RSA>(keyPem, password, RSA.Create, certificate.CopyWithPrivateKey),
                     Oids.Dsa => ExtractKeyFromEncryptedPem<DSA>(keyPem, password, DSA.Create, certificate.CopyWithPrivateKey),
-                    Oids.EcPublicKey => ExtractKeyFromEncryptedPem<ECDsa>(keyPem, password, ECDsa.Create, certificate.CopyWithPrivateKey),
+                    Oids.EcPublicKey when IsECDsa(certificate) =>
+                        ExtractKeyFromEncryptedPem<ECDsa>(
+                            keyPem,
+                            password,
+                            ECDsa.Create,
+                            certificate.CopyWithPrivateKey),
+                    Oids.EcPublicKey when IsECDiffieHellman(certificate) =>
```

Comment:
I think flipping these will fix the Windows problems. IIRC, ECDH keys can do ECDH or ECDSA, but ECDSA keys can't do ECDH.

Reply:
Is it even possible to get into this second EcPublicKey block on any OS?  For Linux and macOS it's just an ECC key.  For Windows I think the only thing that might stand a chance of failing this test is a Montgomery curve (Curve25519), since (IIRC) Montgomery curves are only allowed to do ECDH on Windows (I don't remember if that fails at import time, or signing time).

Reply:
The `GetECDsaPublicKey` and friends take in to account of the certificate's key usage. So if the cert has a keyUsage of only `keyAgreement`, `GetECDsaPublicKey` will return null and we'll end up in here.